src/features/repeatable.tsx

```diff
@@ -212,14 +221,17 @@ export function createRepeatable<T extends RepeatableOptions>(
                         {currDisplay.showAmount === false ? null : (
                             <div>
                                 <br />
-                                {unref(genericRepeatable.limit) === Decimal.dInf ? (
-                                    <>Amount: {formatWhole(genericRepeatable.amount.value)}</>
-                                ) : (
-                                    <>
-                                        Amount: {formatWhole(genericRepeatable.amount.value)} /{" "}
-                                        {formatWhole(unref(genericRepeatable.limit))}
-                                    </>
-                                )}
+                                joinJSX(
+                                    <>Amount: {formatWhole(genericRepeatable.amount.value)}</>,
+                                    {unref(genericRepeatable.limit) !== Decimal.dInf ? (
+                                        <> / {formatWhole(unref(genericRepeatable.limit))}</>
+                                    ) : undefined},
+                                    {(genericRepeatable as GenericRepeatable & GenericBonusAmountFeature).bonusAmount == null ? null : (
```

Comment:
I feel the decorators shouldn't have to be coded into the feature themselves if at all possible. I feel like the great potential of this feature would be being able to distribute the decorator file itself to other devs and it being fully self contained.

I think implementing a generic way for decorators to modify JSX functions should probably take form in two different ways: A way to wrap the JSX function (so the decorator is expected to render the JSX function somewhere inside it's own), as well as a way for each decorator to add a JSX function that the default JSX function is expected to render somewhere (probably at the end) - similar to how the options menu has options injected in when features are used.

Reply:
Yeah, I put this here because you wanted it to show the bonus amount in the feature if possible, but in the end, this was the only feature that was doable in. I'll try out having insertion points for the JSX functions, although it might just be a thing where if you want to use them, you should be either writing your own full display function or even swapping out the component (which is what the `getGatheredProps` function is for).

Reply:
Are you still planning on implementing this for this PR, or do you think it belongs in a separate PR?

Reply:
Just getting the rest of the stuff sorted out as best I can before focusing on this. *Probably* going to need to push it to its own thing, but there's a chance it's at least partially doable without overhauling how displays are built.